harness/determined/common/experimental/checkpoint/_checkpoint.py

```diff
@@ -112,7 +116,7 @@ def _find_shared_fs_path(self, checkpoint_storage: Dict[str, Any]) -> pathlib.Pa
             "checkpoint storage configuration.".format(self.uuid, potential_paths)
         )
 
-    def download(self, path: Optional[str] = None) -> str:
+    def download(self, path: Optional[str] = None, via_master: bool = False) -> str:
```

Comment:
There's three different behaviors we want to support: auto, via master, and direct, right?  As you said, there will be times that the user wants to control via-master-vs-direct, so a bool isn't enough to express the level of control we want to give the user.

Probably it's time for an enum here, that defaults to AUTO.

Reply:
Didn't think about direct-only, but I can add it.